modules/core/04-channel/keeper/upgrade.go

```diff
@@ -363,6 +363,27 @@ func (k Keeper) WriteUpgradeAckChannel(ctx sdk.Context, portID, channelID string
 	emitChannelUpgradeAckEvent(ctx, portID, channelID, channel, upgrade)
 }
 
+func (k Keeper) WriteUpgradeConfirmChannel(ctx sdk.Context, portID, channelID string, counterpartyUpgrade types.Upgrade) {
+	defer telemetry.IncrCounter(1, "ibc", "channel", "upgrade-confirm")
+
+	channel, found := k.GetChannel(ctx, portID, channelID)
+	if !found {
+		panic(fmt.Sprintf("could not find existing channel when updating channel state in successful ChanUpgradeAck step, channelID: %s, portID: %s", channelID, portID))
+	}
+
+	if !k.HasInflightPackets(ctx, portID, channelID) {
+		previousState := channel.State
+		channel.State = types.STATE_FLUSHCOMPLETE
+		k.SetChannel(ctx, portID, channelID, channel)
+
+		k.Logger(ctx).Info("channel state updated", "port-id", portID, "channel-id", channelID, "previous-state", previousState, "new-state", channel.State)
+	} else {
+		k.SetCounterpartyUpgrade(ctx, portID, channelID, counterpartyUpgrade)
```

Comment:
We should only need the counterparty upgrade timeout, right? Do we have another issue where we will refactor the store from counterparty upgrade to counterparty upgrade timeout?

Reply:
Ah, just saw [this comment](https://github.com/cosmos/ibc-go/issues/4233#issuecomment-1662336935) when going through emails. So all good. :)